### PR TITLE
DEV-345: Sit V2.5

### PR DESCRIPTION
**Pull request by PrestonB1123**
updated sit to version 2.5 that utilizes the avatar api and no longer utilizes zone entities.
https://highfidelity.atlassian.net/browse/DEV-345

### CODE REVIEW

File: usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js [hunk 1]
```diff
@@ -92,7 +92,7 @@
             if (DEBUG) {
                 console.log("MOUSE RELEASE");
             }
-            if (event.isPrimaryButton) {
+            if (event.isPrimaryButton && Vec3.distance(MyAvatar.position,Entities.getEntityProperties(_this.entityID,["position"]).position) <= 5) {
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js]:
```suggestion
            if (event.isPrimaryButton &&
                Vec3.distance(MyAvatar.position, Entities.getEntityProperties(_this.entityID,["position"]).position) <= MAX_CLICK_DISTANCE_M) {
```

and then add a `MAX_CLICK_DISTANCE_M` var at the top of the file, perhaps right below line 20.

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 2]
```diff
@@ -222,6 +163,7 @@
                 HMD.displayModeChanged.connect(_this.standUpWrapper);
                 _this.connectedSignals = true;
             }
+
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: remove extra whitespace please

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 3]
```diff
@@ -362,75 +296,15 @@
...
-            for (var i in DISABLED_DRIVE_KEYS_DURING_SIT) {
-                MyAvatar.enableDriveKey(DISABLED_DRIVE_KEYS_DURING_SIT[i]);
-            }
-            driveKeysDisabled = false;
-        }
+        MyAvatar.endSit(MyAvatar.position, MyAvatar.orientation);
+        createSittableUI();
     }
-
+
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: remove extra whitespace please

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 4]
```diff
@@ -601,10 +471,9 @@
             _this.sittableID = false;
         }
     }
-
+
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: remove extra whitespace please

File: usefulUtilities/sitV2/appResources/appData/sitServerV2.js [hunk 5]
```diff
@@ -184,8 +184,8 @@
         _this.entityID = id;
         _this.isOccupied = false;
         _this.resolved = false;
-        createCanSitZone();
-        setNextTimeout();
+        //createCanSitZone();
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitServerV2.js]: delete commented lines

File: usefulUtilities/sitV2/appResources/appData/sitServerV2.js [hunk 6]
```diff
@@ -196,7 +196,7 @@
             Script.clearInterval(_this.heartbeatInterval);
             _this.heartbeatInterval = false;
         }
-        deleteCanSitZone();
+        //deleteCanSitZone();
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitServerV2.js]: delete commented line

File: usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js (hunk 1 above)
Comment (PrestonB1123) [reply usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js]: Ack

File: usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js [hunk 7]
```diff
@@ -18,6 +18,7 @@
     var SITTABLE_ALPHA_DELTA = 0.009;
     var SITTABLE_FADE_MS = 40; // "Click/Trigger to Sit" local entity image fade after 50 ms
     var TIMEOUT_BEFORE_FADE_MS = 1000;
+    MAX_SIT_DISTANCE_M = 5;
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js]:
```suggestion
    var MAX_SIT_DISTANCE_M = 5;
```

File: usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js [hunk 8]
```diff
@@ -92,7 +93,7 @@
             if (DEBUG) {
                 console.log("MOUSE RELEASE");
             }
-            if (event.isPrimaryButton) {
+            if (event.isPrimaryButton && Vec3.distance(MyAvatar.position,Entities.getEntityProperties(_this.entityID,["position"]).position) <= MAX_SIT_DISTANCE_M) {
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js]: please split up to multiple lines with proper indentation (as per my earlier suggestion)

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 9]
```diff
@@ -221,7 +162,7 @@
                 MyAvatar.wentAway.connect(_this.standUpWrapper);
                 HMD.displayModeChanged.connect(_this.standUpWrapper);
                 _this.connectedSignals = true;
-            }
+            }
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: please remove extra whitespace

File: usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js [hunk 10]
```diff
@@ -92,7 +93,8 @@
             if (DEBUG) {
                 console.log("MOUSE RELEASE");
             }
-            if (event.isPrimaryButton) {
+            if (event.isPrimaryButton && 
+                Vec3.distance(MyAvatar.position,Entities.getEntityProperties(_this.entityID,["position"]).position) <= MAX_SIT_DISTANCE_M) {
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/resources/sittableUIClient.js]:
```suggestion
                Vec3.distance(MyAvatar.position, Entities.getEntityProperties(_this.entityID,["position"]).position) <= MAX_SIT_DISTANCE_M) {
```

**Pull request merged**